contracts/pools/PoolCollection.sol

```diff
@@ -624,15 +627,42 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
         bytes32 contextId,
         address provider,
         Token pool,
-        uint256 poolTokenAmount
-    ) external only(address(_network)) validAddress(provider) greaterThanZero(poolTokenAmount) returns (uint256) {
+        uint256 poolTokenAmount,
+        uint256 reserveTokenAmount
+    )
+        external
+        only(address(_network))
+        validAddress(provider)
+        greaterThanZero(poolTokenAmount)
+        greaterThanZero(reserveTokenAmount)
+        returns (uint256)
+    {
         Pool storage data = _poolStorage(pool);
+        PoolLiquidity memory liquidity = data.liquidity;
+
+        uint256 poolTokenTotalSupply = data.poolToken.totalSupply();
+        uint256 underlyingAmount = _poolTokenToUnderlying(
+            poolTokenAmount,
+            poolTokenTotalSupply,
+            liquidity.stakedBalance
+        );
+
+        if (reserveTokenAmount > underlyingAmount) {
+            revert InvalidParam();
+        }
 
         // obtain the withdrawal amounts
-        InternalWithdrawalAmounts memory amounts = _poolWithdrawalAmounts(pool, data, poolTokenAmount);
+        InternalWithdrawalAmounts memory amounts = _poolWithdrawalAmounts(
+            pool,
+            poolTokenAmount,
+            reserveTokenAmount,
```

Comment:
If I understand correctly, we previously used the entire `underlyingAmount` here.
Now we are using the possibly smaller `reserveTokenAmount` instead.
Shouldn't the diff `underlyingAmount - reserveTokenAmount` be dealt with?

Reply:
We pass the exact TKN that was calculated during the initiation of the withdrawal request